etrago/cluster/electrical.py

```diff
@@ -645,14 +648,20 @@ def unify_foreign_buses(etrago):
     return busmap_foreign
 
 
-def preprocessing(etrago, apply_on="grid_model"):
+def preprocessing(
+    etrago, used_for="spatial_clustering", apply_on="grid_model"
+):
     """
     Preprocesses an Etrago object to prepare it for network clustering.
 
     Parameters
     ----------
     etrago : Etrago
         An instance of the Etrago class
+    used_for : string
+        provide information about the objective of the preprocessing. Which
+        process is going to use the result. e.g. "spatial_clustering",
+        "market_model".
```

Comment:
This parameter seems to be very similar to the other new parameter "apply_on" that is already in `dev`. 
Is the "used_for" parameter always "market_model" when applied on the market model and "spatial_clustering" if it is applied on the grid_model /etrago.network?

Reply:
Yes, it is. I will adjust my PR to work using the already existing parameter "apply_on".